database/migrations/2016_02_17_175810_create_competition_user_pivot_table.php

```diff
@@ -13,9 +13,9 @@ class CreateCompetitionUserTable extends Migration
     public function up()
     {
         Schema::create('competition_user', function (Blueprint $table) {
-            $table->integer('competition_id')->unsigned()->index();
+            $table->integer('competition_id')->unsigned();
             $table->foreign('competition_id')->references('id')->on('competitions');
-            $table->string('user_id')->index();
+            $table->string('user_id');
```

Comment:
Yep, this was the big reason I was getting a lot of :rotating_light: errors when trying to associate pivot tables. If designating a foreign key, Laravel already creates an index and names it `whatever_whatever_foreign` and looks for that when making the association. By specifying `->index()` it would change the name of the index key to `whatever_whatever_index` and Laravel would get angry cause it's expecting it to end in `_foreign`. This was the initial issue causing a bunch of errors with trying to get the pivot tables to work. The second was the `$incrementing = true` issue.